yt/frontends/enzo_p/data_structures.py

```diff
@@ -0,0 +1,385 @@
+        self.storage_filename = storage_filename
+        Dataset.__init__(self, filename, dataset_type, file_style=file_style,
+                         units_override=units_override, unit_system=unit_system)
+
+    def _parse_parameter_file(self):
+        """
+        Parses the parameter file and establishes the various
+        dictionaries.
+        """
+
+        f = open(self.parameter_filename, "r")
+        # get dimension from first block name
+        b0, fn0 = f.readline().strip().split()
+        level0, left0, right0 = get_block_info(b0, min_dim=0)
+        root_blocks = get_root_blocks(b0)
+        f.close()
+        self.dimensionality = left0.size
+        self.periodicity = \
+          ensure_tuple(np.ones(self.dimensionality, dtype=bool))
+
+        fh = h5py.File(os.path.join(self.directory, fn0), "r")
+        self.domain_left_edge  = fh.attrs["lower"]
+        self.domain_right_edge = fh.attrs["upper"]
+
+        # all blocks are the same size
+        ablock = fh[list(fh.keys())[0]]
+        self.current_time = ablock.attrs["time"][0]
+        gsi = ablock.attrs["enzo_GridStartIndex"]
+        gei = ablock.attrs["enzo_GridEndIndex"]
+        self.ghost_zones = gsi[0]
+        self.root_block_dimensions = root_blocks
+        self.active_grid_dimensions = gei - gsi + 1
+        self.grid_dimensions = ablock.attrs["enzo_GridDimension"]
+        self.domain_dimensions = root_blocks * self.active_grid_dimensions
+        fh.close()
+
+        self.periodicity += (False, ) * (3 - self.dimensionality)
+
+        # WIP hard-coded for now
+        self.refine_by = 2
+        self.cosmological_simulation = 0
+        self.gamma = 5. / 3.
+        self.particle_types = ()
+        self.particle_types_raw = self.particle_types
+        self.unique_identifier = \
+          str(int(os.stat(self.parameter_filename)[stat.ST_CTIME]))
+
+    def _set_code_unit_attributes(self):
+        setdefaultattr(self, 'length_unit', self.quan(1, "cm"))
+        setdefaultattr(self, 'mass_unit', self.quan(1, "g"))
+        setdefaultattr(self, 'time_unit', self.quan(1, "s"))
+        setdefaultattr(self, 'velocity_unit',
+                       self.length_unit / self.time_unit)
+        magnetic_unit = np.sqrt(4*np.pi * self.mass_unit /
+                                (self.time_unit**2 * self.length_unit))
+        magnetic_unit = np.float64(magnetic_unit.in_cgs())
+        setdefaultattr(self, 'magnetic_unit',
+                       self.quan(magnetic_unit, "gauss"))
+
+    @classmethod
+    def _is_valid(cls, *args, **kwargs):
+        fn = args[0]
+        ddir = os.path.dirname(fn)
+        if not fn.endswith(".block_list"):
+            return False
+        try:
+            with open(fn, "r") as f:
+                block, block_file = f.readline().strip().split()
+                get_block_info(block)
+                if not os.path.exists(os.path.join(ddir, block_file)):
+                    return False
+        except:
```

Comment:
Can this be `except Exception`. The way it is now will catch things like `KeyboardInterrupt` and `SystemExit`.